### ISSUE TITLE
Text: phantom square appears when using SSAO 

### ISSUE THREAD

**Issue by petemesquite** [protectwise/troika#63]
First of all, thank you so much for sharing this library!

I'm using the standard SSAO and I'm passing a material that isn't writing to the depth buffer, but I'm still seeing what looks like a small quad above where the text is rendering. It seems like it only shows up when the textMesh object is rotated.

![image](https://user-images.githubusercontent.com/7060111/84959220-29cf1980-b0bc-11ea-96c7-13d9197d56ba.png)

The text on the left is scaled up x20 but you can also see it on the right. Am I doing something wrong or is this a limitation?

**Comment by lojjic** [comment]
That's very strange. I haven't had any experience with "the standard SSAO" personally, could you provide more details on that? Ideally a testcase on Codesandbox or similar that I can poke around with?

**Comment by lojjic** [comment]
Assuming this is the SSAOPass from three's postprocessing examples, it looks like it uses `scene.overrideMaterial` to render the scene with a MeshNormalMaterial for every object. This makes sense why you'd see a square there since it's ignoring the TextMesh's customized material with the required shader code to draw the glyphs.

Honestly I don't see how this would work for *any* materials that use custom shaders for transforming vertices or transparency. Is this perhaps a known issue with SSAOPass?

**Comment by petemesquite** [comment]
Yep, it is the SSAOPass. That same implementation is in drei under the StandardEffects component. Here's a [Codesandbox](https://codesandbox.io/s/textmesh-phantom-square-uhdhj?file=/src/index.js)

If you set depthWrite back to true, you'll see the effect around the edges of the font. So it is sort of working... It also only shows up if there is other geometry behind it, if you take away the plane the artifact disappears.

**Comment by lojjic** [comment]
Ah, I see, `drei` uses the external `postprocessing` package, not the one from Three's examples. They're pretty different, but I think the underlying issue of using a scene overrideMaterial may be the same.

I do know that very recent `postprocessing` versions include a [workaround for applying override materials](https://vanruesc.github.io/postprocessing/public/docs/class/src/core/OverrideMaterialManager.js~OverrideMaterialManager.html#static-set-workaroundEnabled) that allows them to work with more advanced materials. I think that if you were to use a more recent version of `postprocessing` you could enable that workaround and it may fix the issue. However you'll also need the latest version of `troika-3d-text` which includes some [bugfixes when used with postprocessing](https://github.com/protectwise/troika/issues/59#issuecomment-641230860)

I briefly tried making that work by upgrading `drei` and `troika-3d-text` in your Codesandbox, but that seemed to break the whole effect even on non-text objects so there may be something else going on. Can you give that a try in your project?

**Comment by petemesquite** [comment]
Oh sorry for the confusion, I am using the external postprocessing package. Upgrading broke the effect in my project too, so there's something else going on now.

Thanks for taking a look at this, I'll keep plugging away at it and report back if I can get something sorted out.

**Comment by lojjic** [comment]
Great, definitely let me know how it goes. I want to add something to the README docs about this postprocessing use case so your findings will help with that.

**Comment by petemesquite** [comment]
Okay, looks like the SSAO effect was recently updated as well, I just had to change some of the attribute values (by orders of magnitude). Once I got that dialed in, the square issue was actually a lot less pronounced but it was still there. I enabled the workaround and that totally solved it! 

I did notice that if I tried to use the `Text` component from `drei` there was a cascade of errors and the frame rate tanks, but I assume there's just some old version of something causing that.

Thanks for your help!
